crates/block-producer/src/batch_builder/batch.rs

```diff
@@ -81,17 +81,21 @@ impl TransactionBatch {
     ///   in the batch.
     /// - Hashes for corresponding input notes and output notes don't match.
     #[instrument(target = "miden-block-producer", name = "new_batch", skip_all, err)]
-    pub fn new(
-        txs: Vec<ProvenTransaction>,
+    pub fn new<'a, I>(
+        txs: impl IntoIterator<Item = &'a ProvenTransaction, IntoIter = I>,
         found_unauthenticated_notes: NoteAuthenticationInfo,
-    ) -> Result<Self, BuildBatchError> {
-        let id = Self::compute_id(&txs);
+    ) -> Result<Self, BuildBatchError>
+    where
+        I: Iterator<Item = &'a ProvenTransaction> + Clone,
```

Comment:
Doc comments for this function need to be updated (e.g., line 73 has a reference to a vector of proven transactions). I would also probably write a brief explanation why the types are defined as they are (my understanding is that we can cheaply clone the iterator we get from `txs.into_iter()`, right?).

Reply:
Correct; we're essentially looking to get `T: Iterator<Item = ProvenTransaction> + Clone` which happens to be fulfilled by the above.